redfish-core/include/utils/hw_isolation.hpp

```diff
@@ -0,0 +1,343 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace redfish
+{
+namespace hw_isolation_utils
+{
+
+constexpr auto objMapperSerName = "xyz.openbmc_project.ObjectMapper";
+constexpr auto objMapperObjPath = "/xyz/openbmc_project/object_mapper";
+constexpr auto objMapperIface = "xyz.openbmc_project.ObjectMapper";
+
+constexpr auto hwIsolationRootDBusPath =
+    "/xyz/openbmc_project/hardware_isolation";
+constexpr auto hwIsolationCreateIface =
+    "xyz.openbmc_project.HardwareIsolation.Create";
+
+/**
+ * @brief API used to isolate the given resource
+ *
+ * @param[in] aResp - The redfish response to return to the caller.
+ * @param[in] resourceName - The redfish resource name which trying to isolate.
+ * @param[in] resourceId - The redfish resource id which trying to isolate.
+ * @param[in] resourceObjPath - The redfish resource dbus object path.
+ * @param[in] hwIsolationDbusName - The HardwareIsolation dbus name which is
+ *                                  hosting isolation dbus interfaces.
+ *
+ * @return The redfish response in given response buffer.
+ *
+ * @note This function will return the appropriate error based on the isolation
+ *       dbus "Create" interface error.
+ */
+inline void
+    isolateResource(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+                    const std::string& resourceName,
+                    const std::string& resourceId,
+                    const sdbusplus::message::object_path& resourceObjPath,
+                    const std::string& hwIsolationDbusName)
+{
+    crow::connections::systemBus->async_method_call(
+        [aResp, resourceName, resourceId,
+         resourceObjPath](const boost::system::error_code ec,
+                          const sdbusplus::message::message& msg) {
+            if (!ec)
+            {
+                messages::success(aResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value() << " : "
+                             << ec.message()
+                             << "] when tried to isolate the given resource: "
+                             << resourceObjPath.str;
+
+            const sd_bus_error* dbusError = msg.get_error();
+            if (dbusError == nullptr)
+            {
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            BMCWEB_LOG_ERROR << "DBus ErrorName: " << dbusError->name
+                             << " ErrorMsg: " << dbusError->message;
+
+            if (strcmp(dbusError->name, "xyz.openbmc_project.Common.Error."
+                                        "InvalidArgument") == 0)
+            {
+                constexpr bool isolate = false;
+                messages::propertyValueIncorrect(aResp->res, "@odata.id",
+                                                 std::to_string(isolate));
```

Comment:
Why the false here? Why not pass the string the user passed? 
https://github.com/openbmc/bmcweb/blob/ed3982131dcef2b499da36e674d2d21b2289ef29/redfish-core/lib/metric_report_definition.hpp#L197